### PR TITLE
Notifications: Microsoft Teams

### PR DESCRIPTION
**Pull request by mikenz**
Add support to send notifications to MS Teams

image

### CODE REVIEW

**Comment by mikenz** [comment]
IRL:
image

File: pkg/notifications/teams.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func init() {
+	initers = append(initers, func(cfg map[string]string) Notifier {
+		if url, ok := cfg["teams_url"]; ok {
```
Comment (tlimoncelli) [inline pkg/notifications/teams.go:13]: Typical Go style is to return nil early:

    		if url, ok := cfg["teams_url"]; !ok {
    		    return nil
    		}
    		notifier := &teamsNotifier{

...

File: pkg/notifications/teams.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	if preview {
+		payload.Text = fmt.Sprintf("**DnsControl Preview %s**\n%s", domain, msg)
+	} else if err != nil {
+		payload.Text = fmt.Sprintf("**DnsControl Error Making Changes %s**\n%s\nError: %s", domain, msg, err)
+	} else {
+		payload.Text = fmt.Sprintf("**DnsControl Successfully Changed %s**\n%s", domain, msg)
+	}
+
+	json, _ := json.Marshal(payload)
```
Comment (tlimoncelli) [inline pkg/notifications/teams.go:46]: Please check the error and return it upstream.

File: pkg/notifications/teams.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	http.Post(s.URL, "text/json", bytes.NewReader(json))
```
Comment (tlimoncelli) [inline pkg/notifications/teams.go:47]: I think `http.Post` returns an error. Please check.

File: pkg/notifications/teams.go (hunk 2 above)
Comment (mikenz) [reply pkg/notifications/teams.go:46]: If I return it upstream I'm going to have to change the interface definition as it doesn't currently allow for this. I guess this is going to break anyone else's custom notify types, are you okay with that? 

If so, `Notify(domain, provider string, message string, err error, preview bool) (err error)`, preferences on naming for the two err's?

Comment (tlimoncelli) [reply pkg/notifications/teams.go]: Oops... I misread the signature.  (I should get more sleep :-) )

I'll file a bug to improve this in the future.

File: pkg/notifications/teams.go (hunk 1 above)
Comment (mikenz) [reply pkg/notifications/teams.go]: Fixed and branch squashed

**Pull request merged**